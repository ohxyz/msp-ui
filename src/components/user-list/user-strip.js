const util = require( '../core/util.js' );
const React = require( 'react' );
const UserProfile = require( '../../model/user-profile.js' ).UserProfile;

function getExpandIconText( shouldExpand ) {

    return shouldExpand === true ? 'remove' : 'add';

}

class UserStrip extends React.Component {

    constructor( props ) {

        super( props );

        this.handleExpandClick = this.handleExpandClick.bind( this );

        this.state = {

            user: null,
            shouldDropDownExpanded: false,
            expandIconText: 'add'
        };
    }

    static getDerivedStateFromProps( nextProps, prevState ) {

        let shouldExpand = util.setDefault( nextProps.shouldDropDownExpanded, false );
        
        return {

            user: util.setDefault( new UserProfile( nextProps.user ), new UserProfile() ),
            shouldDropDownExpanded: shouldExpand,
            expandIconText: getExpandIconText( shouldExpand )
        };
    }

    handleExpandClick() {

        let shouldExpand = !this.state.shouldDropDownExpanded;

        this.setState( { 

            shouldDropDownExpanded: shouldExpand,
            expandIconText: getExpandIconText( shouldExpand )

        } );
    }

    renderDropDown() {

        if ( this.state.shouldDropDownExpanded === true ) {

            return (

                <div className="user-strip__dropdown">
                    <div className="user-strip__dropdown__content">User access level</div>
                </div>
            );
        }
    }

    render() {

        let user = this.state.user;

        return (

            <div className="user-strip">
                <div className="user-strip__main">
                    <div className="user-strip__delete material-icons">close</div>
                    <div 
                        className="user-strip__expand material-icons" 
                        onClick={ this.handleExpandClick }
                    >
                        { this.state.expandIconText }
                    </div>
                    <div className="user-strip__top-bar">
                        <span className="user-strip__full-name">{ user.fullName }</span>
                        <span className="user-strip__top-group">{ user.topGroup }</span>
                    </div>
                    <div className="user-strip__middle-bar">
                        <div className="user-strip__group">{ user.group }</div>
                    </div>
                </div>
                { this.renderDropDown() }
            </div>
        );
    }
}


module.exports = {

    UserStrip,
};

