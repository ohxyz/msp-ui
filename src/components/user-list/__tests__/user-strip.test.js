import React from 'react';
import { UserStrip } from '../user-strip.js';
import ReactTestRender from 'react-test-renderer';

it( 'should render with defaults', () => {

    let user = { 

        fullName: 'Full Name',
        topOrgName: 'Top Orgnisation Name',
        orgName: 'Orgnisation Name'
    };
    
    const tree = ReactTestRender
        .create( <UserStrip user={ user }/> )
        .toJSON();

    expect( tree ).toMatchSnapshot();

} );


describe( 'UserStrip instance', () => { 

    let userStrip = new UserStrip();
    let mockFn = jest.fn();

    userStrip.setState = mockFn;

    beforeEach(() => {

        mockFn.mockClear();
    });

    test( "changes it's state after calling handleChevronIconClick", () => { 

        userStrip.handleChevronIconClick();
        expect( mockFn.mock.calls.length ).toBe( 1 );

    } );

    test( "changes it's state after calling handleDeleteIconClick", () => { 

        userStrip.handleDeleteIconClick();
        expect( mockFn.mock.calls.length ).toBe( 1 );

    } );

    test( "change's its state after calling No button is clicked ", () => { 

        userStrip.state.onPropsDeleteUserYesClick = mockFn;
        userStrip.handleDeleteUserNoButtonClick();

        expect( mockFn.mock.calls.length ).toBe( 1 );
    } );

    test( "change's its state after calling Delete Icon is clicked ", () => { 

        userStrip.handleDeleteIconClick();

        expect( mockFn.mock.calls.length ).toBe( 1 );
        
    } );

} );