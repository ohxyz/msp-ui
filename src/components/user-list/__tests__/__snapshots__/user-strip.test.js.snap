// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with defaults 1`] = `
<div
  className="user-strip"
>
  <div
    className="user-strip__main"
  >
    <div
      className="user-strip__delete material-icons"
      onClick={[Function]}
    >
      delete_forever
    </div>
    <div
      className="user-strip__expand material-icons"
      onClick={[Function]}
    >
      keyboard_arrow_down
    </div>
    <div
      className="user-strip__top-bar"
    >
      <span
        className="user-strip__full-name"
      >
        Full Name
      </span>
      <span
        className="user-strip__top-org"
      >
        Top Orgnisation Name
      </span>
    </div>
    <div
      className="user-strip__middle-bar"
    >
      <div
        className="user-strip__org"
      >
        Orgnisation Name
      </div>
    </div>
  </div>
</div>
`;
